### ISSUE TITLE
SIFT normalization and the pre-trained vocabulary trees

### ISSUE THREAD

**Issue by an-kumar** [colmap/colmap#578]
Colmap distributes vocabulary trees for retrieval that have been trained offline.

I was wondering if options in SIFT extraction such as normalization, and the various other options, are relevant w.r.t using the offline-trained vocabulary tree. If we use different options than what was used to train the tree, performance will degrade I assume.

Similar question w.r.t GPU vs CPU, as well as using DSP-Sift vs not.

**Comment by ahojnnes** [comment]
Yes, the trees were trained for default SIFT parameters (non-DSP, non-affine). Both CPU and GPU features should be compatible and there shouldn’t be any problem in mixing them, since I am converting them to a common format. Apart from the DSP and affine options, you should be fine to change the parameters and use the pretrained trees. I will actually update the website to highlight this fact. Thanks for the question.

**Comment by an-kumar** [comment]
I would think the normalization option would also not be change-able (L1 vs L2), do you agree?

**Comment by ahojnnes** [comment]
Yes, indeed normalization will also have an impact. But I would always stick with L1 root normalization anyway, which was used to train the trees.
